code/controllers/CMSMain.php

```diff
@@ -710,6 +713,20 @@ public function getEditForm($id = null, $fields = null) {
 		}
 	}
 
+	/**
+	 * @param SiteTree $record
+	 */
+	protected function getArchiveWarningMessage($record)
+	{
+		// Get all page descendants
+		$record->collateDescendants(true, $descendants);
```

Comment:
@tractorcow mentioned if we could use `numChildren` instead, do you think that's still a reasonable change to have?